apps/api/views/metrics.py

```diff
@@ -0,0 +1,750 @@
+# top row of the CSV document.
+
+class AdminSubmissionCSVRenderer(renderers.CSVRenderer):
+    labels = {
+        'date': 'Date',
+        'submissions_made_count': 'Submissions Made',
+        'submissions_total': 'Submissions Total',
+        'score_interval': 'Score Interval',
+        'score_interval_count': 'Scores on Interval',
+    }
+    header = list(labels.keys())
+
+
+class AdminKlassCSVRenderer(renderers.CSVRenderer):
+    labels = {
+        'date': 'Date',
+        'klasses_created_count': 'Classes Created',
+        'klasses_total': 'Classes Total',
+        'ave_students_per_klass': 'Average Number of Students Per Class',
+        'ave_definitions_per_klass': 'Average Number of Homework Definitions Per Class',
+        'ave_submissions_per_definition': 'Average Number of Submissions Per Homework Definitions',
+    }
+    header = list(labels.keys())
+
+
+class AdminUserCSVRenderer(renderers.CSVRenderer):
+    labels = {
+        'date': 'Date',
+        'students_join_count': 'Students Joined',
+        'instructors_join_count': 'Instructors Joined',
+        'students_total': 'Students Total',
+        'instructors_total': 'Instructors Total',
+        'users_total': 'Users Total',
+    }
+    header = list(labels.keys())
+
+
+class InstructorKlassCSVRenderer(renderers.CSVRenderer):
+    labels = {
+        'name': 'Homework Name',
+        'score': 'Average Homework Score',
+        'time': 'Time of Day',
+        'count': 'Submission Count',
+    }
+    header = list(labels.keys())
+
+
+class InstructorStudentCSVRenderer(renderers.CSVRenderer):
+    labels = {
+        'name': 'Homework Name',
+        'score': 'Homework Score',
+        'time': 'Time of Day',
+        'count': 'Submission Count',
+    }
+    header = list(labels.keys())
+
+
+class InstructorTeamCSVRenderer(renderers.CSVRenderer):
+    labels = {
+        'name': 'Homework Name',
+        'score': 'Homework Score',
+        'time': 'Time of Day',
+        'count': 'Submission Count',
+        'cha_username': 'Chagrade Username',
+        'submission_count': 'Submission Count',
+        'commit_count': 'Github Commit Count',
+    }
+    header = list(labels.keys())
+
+
+def list_of_dicts_to_dict_of_lists(l):
+    return {k: [d[k] for d in l] for k in l[0]}
+
+
+def merge_list_of_lists_of_dicts(input_list):
+    # Sort list by lengths of the sub-lists from greatest length to shortest length
+    sorted_input_list = sorted(input_list, key=lambda i: -len(i))
+    output_list = []
+    if len(sorted_input_list) > 0:
+        output_list = deepcopy(sorted_input_list.pop(0))
+    else:
+        return []
+    while len(sorted_input_list) > 0:
+        next_longest_list = sorted_input_list.pop(0)
+        for i in range(len(next_longest_list)):
+            output_list[i].update(next_longest_list[i])
+    return output_list
+
+
+def union_lists(union_key, list1_data, list2_data):
+    if not list1_data.get('sorted'):
```

Comment:
Could we use more descriptive variable names here? This is a little hard to read at first glance

Reply:
I changed `union_key` to `common_key`, to make it more clear. I don't think variable names are the real problem. I think it may just be unclear what this algorithm's purpose is. I left a comment in the function. Let me know if that clears things up for you.

Reply:
I think we already talked about this, but I was more referring to l1/l2 type stuff.

Reply:
Done.